jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/LockedPool.java

```diff
@@ -0,0 +1,183 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.util.Collection;
+import java.util.function.Function;
+import java.util.function.ToIntFunction;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.jetty.util.thread.AutoLock;
+
+public class LockedPool<P> implements Pool<P>
```

Comment:
This should be `LockedConcurrentPool` or `ConcurrentPool.Locked`  or it should have the pool that it wraps passed in.   Also needs javadoc.

Reply:
Converted to a `Pool.Wrapper`, but left the name as `LockedPool` as `LockedPoolWrapper` seemed a mouthful.